app/src/features/rules/components/RulesList/components/RulesListIndex/RulesListIndex.tsx

```diff
@@ -1,39 +1,314 @@
+                ) : null}
+              </Menu.Item>
+            </PremiumFeature>
+          ))}
+      </Menu>
+    );
+  }, [handleNewRuleOnClick, getFeatureLimitValue]);
+
+  const handleNewGroupClick = () => {
+    setIsCreateNewRuleGroupModalActive(true);
+  };
+
+  const buttonData = [
+    {
+      type: "text",
+      isTooltipShown: true,
+      tourId: "rule-table-create-group-btn",
+      buttonText: "New Group",
+      icon: <RiFolderAddLine className="anticon" />,
+      onClickHandler: handleNewGroupClick,
+    },
+    {
+      isTooltipShown: true,
+      hasPopconfirm: true,
+      buttonText: "Import",
+      authSource: AUTH.SOURCE.UPLOAD_RULES,
+      icon: <DownloadOutlined />,
+      onClickHandler: handleImportRulesOnClick,
+      trackClickEvent: () => {
+        trackUploadRulesButtonClicked(AUTH.SOURCE.RULES_LIST);
+      },
+    },
+    {
+      type: "primary",
+      isTooltipShown: false,
+      buttonText: "New Rule",
+      icon: <DownOutlined />,
+      onClickHandler: handleNewRuleOnClick,
+      isDropdown: true,
+      overlay: dropdownOverlay,
+    },
+  ];
+
+  const contentHeaderActions = buttonData.map(
+    (
+      { icon, type = null, buttonText, isTooltipShown, onClickHandler, isDropdown = false, overlay, tourId = null },
+      index
+    ) => (
+      <Tooltip key={index} title={isTooltipShown ? buttonText : null}>
+        <>
+          {isDropdown ? (
+            <Dropdown.Button
+              icon={icon}
+              type={type as DropdownButtonType}
+              trigger={["click"]}
+              onClick={() => onClickHandler()}
+              overlay={overlay}
+              data-tour-id={tourId}
+              className="rule-selection-dropdown-btn"
+            >
+              {buttonText}
+            </Dropdown.Button>
+          ) : (
+            // @ts-ignore
+            <AuthPopoverButton isLoggedIn={user?.details?.isLoggedIn} {...buttonData[index]} />
+          )}
+        </>
+      </Tooltip>
+    )
+  );
 
   return (
-    <RulesListProvider>
+    <>
       <div className="rq-rules-list-container">
         {/* TODO: Add Feature Limiter Banner Here */}
+        {isImportRulesModalActive ? (
+          <ImportRulesModal
+            isOpen={isImportRulesModalActive}
+            toggle={() => setIsImportRulesModalActive((prev) => !prev)}
+          />
+        ) : null}
+
+        {isCreateNewRuleGroupModalActive ? (
+          <CreateNewRuleGroupModal
+            isOpen={isCreateNewRuleGroupModalActive}
+            toggle={() => setIsCreateNewRuleGroupModalActive((prev) => !prev)}
+          />
+        ) : null}
 
         <ContentHeader
           title="My Rules"
           subtitle="Create and manage your rules from here"
-          actions={[<Button type="primary">New Rule</Button>]}
+          actions={contentHeaderActions}
+          searchValue={searchValue}
+          setSearchValue={setSearchValue}
+          activeFilter={activeFilter}
+          filters={[
```

Comment:
Store this this array of filters in  a var maybe named like `headerFilters` and pass this to filters prop.